app/models/account.rb

```diff
@@ -0,0 +1,8 @@
+class Account < ApplicationRecord
+  has_many :account_memberships
+  has_many :users, through: :account_memberships
+
+  delegate :name, to: :profile
```

Comment:
Alternatively, we could delegate all missing methods to the concrete account types.

```suggestion
  delegate_missing_to :profile
```